crates/curp/src/client/mod.rs

```diff
@@ -355,6 +389,14 @@ impl ClientBuilder {
         })
     }
 
+    /// Wait for client id
+    async fn wait_for_client_id(&self, state: Arc<state::State>) {
+        while state.client_id() == 0 {
```

Comment:
I think it maybe dangerous to loop checking client_id without a break or error. If some reason that the client id not update, the program will hang on.

The only way to modify client_id is `check_gen_local_client_id`, which is a leader-only function. So what if a follower calls `wait_for_client_id`?

Reply:
Now it will retry for a limited number of times.

Reply:
The place for other clients to modify client id is in `ClientApi::lease_keep_alive`.